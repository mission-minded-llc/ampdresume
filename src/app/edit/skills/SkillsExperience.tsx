"use client";

import { Box } from "@mui/material";
import React from "react";
import { SkillForUserWithSkill } from "@/graphql/getSkillsForUser";
import { SkillItem } from "./SkillItem";
import { groupSkillsForUserByYearExperience } from "openresume-theme";

export const SkillsExperience = ({ skills }: { skills: SkillForUserWithSkill[] }) => {
  const skillsExperienceList = groupSkillsForUserByYearExperience(skills);

  return (
    <Box
      sx={{
        display: "grid",
        gridTemplateColumns: "auto 1fr",
        gridTemplateRows: "auto",
        width: "100%",
        gap: "5px",
        marginTop: "5px",
      }}
    >
      {skillsExperienceList.map((skillsExperience) => {
        const totalYears = skillsExperience[0];
        const skillsList = skillsExperience[1];

        return (
          <React.Fragment key={`skill-group-${skillsExperience[0]}`}>
            <Box
              sx={{
                fontWeight: "bold",
                padding: "4px 14px 4px 0",
                textAlign: "right",
              }}
            >
              {totalYears ? `${totalYears} year${parseInt(totalYears, 10) > 1 ? "s" : ""}:` : ""}
            </Box>
            <Box
              sx={{
                display: "flex",
                flexWrap: "wrap",
                gap: "5px",
                alignItems: "center",
              }}
            >
              {skillsList.map((skill) => (
                <SkillItem
                  key={`skill-${skill.skill.name}`}
                  skill={skill as SkillForUserWithSkill}
                />
              ))}
            </Box>
          </React.Fragment>
        );
      })}
    </Box>
  );
};
