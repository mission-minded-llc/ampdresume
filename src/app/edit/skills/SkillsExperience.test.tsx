import "@testing-library/jest-dom";

import { SkillForUser, groupSkillsForUserByYearExperience } from "@ampdresume/theme";
import { render } from "@testing-library/react";
import React from "react";

import { SkillsExperience } from "./SkillsExperience";


jest.mock("@ampdresume/theme", () => ({
  groupSkillsForUserByYearExperience: jest.fn(),
}));

jest.mock("./SkillItem", () => ({
  SkillItem: ({ skill }: { skill: SkillForUser }) => <div>{skill.skill.name}</div>,
}));

describe("SkillsExperience", () => {
  const mockSkills = [
    {
      id: "1",
      userId: "user-id",
      skillId: "skill-js",
      skill: { id: "skill-js", name: "JavaScript", icon: "icon-js" },
      description: "Skill description",
      icon: "icon-js",
      yearStarted: 2015,
      totalYears: 6,
    },
    {
      id: "2",
      userId: "user-id",
      skillId: "skill-ts",
      skill: { id: "skill-ts", name: "TypeScript", icon: "icon-ts" },
      description: "Skill description",
      icon: "icon-ts",
      yearStarted: 2017,
      totalYears: 4,
    },
  ];

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly with grouped skills", () => {
    (groupSkillsForUserByYearExperience as jest.Mock).mockReturnValue([
      ["6", [mockSkills[0]]],
      ["4", [mockSkills[1]]],
    ]);

    const { container, getByText } = render(<SkillsExperience skills={mockSkills} />);

    expect(getByText("6 years:")).toBeInTheDocument();
    expect(getByText("JavaScript")).toBeInTheDocument();
    expect(getByText("4 years:")).toBeInTheDocument();
    expect(getByText("TypeScript")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("renders correctly with no skills", () => {
    (groupSkillsForUserByYearExperience as jest.Mock).mockReturnValue([]);

    const { container } = render(<SkillsExperience skills={[]} />);

    // Expect the container to have an empty div.
    expect(container.firstChild).toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });
});
