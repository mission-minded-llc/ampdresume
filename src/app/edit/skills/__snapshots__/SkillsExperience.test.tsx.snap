// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillsExperience renders correctly with grouped skills 1`] = `
<div>
  <div
    class="MuiBox-root css-fgdvah"
  >
    <div
      class="MuiBox-root css-9y227z"
    >
      6 years:
    </div>
    <div
      class="MuiBox-root css-11dbebx"
    >
      <div>
        JavaScript
      </div>
    </div>
    <div
      class="MuiBox-root css-9y227z"
    >
      4 years:
    </div>
    <div
      class="MuiBox-root css-11dbebx"
    >
      <div>
        TypeScript
      </div>
    </div>
  </div>
</div>
`;

exports[`SkillsExperience renders correctly with no skills 1`] = `
<div>
  <div
    class="MuiBox-root css-fgdvah"
  />
</div>
`;
